internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,284 @@
+func (rp *RegistryPod) Create(ctx context.Context) (*corev1.Pod, string, error) {
+	err := rp.validate()
+	if err != nil {
+		return nil, "", fmt.Errorf("error in validating RegistryPod struct: %v", err)
+	}
+
+	pod, err := rp.create(ctx)
+	if err != nil {
+		podLogs, logErr := rp.FetchPodLogs(ctx, pod)
+		if logErr != nil {
+			return nil, podLogs, fmt.Errorf("error creating pod: %v: and fetching logs: %v", err, logErr)
+		}
+		return nil, podLogs, fmt.Errorf("error creating pod: %v", err)
+	}
+
+	err = rp.verifyPodRunning(ctx, pod)
+	if err != nil {
+		podLogs, logErr := rp.FetchPodLogs(ctx, pod)
+		if logErr != nil {
+			return nil, podLogs, fmt.Errorf("error verifying pod creation: %v: and fetching logs: %v", err, logErr)
+		}
+		return nil, podLogs, fmt.Errorf("error verifying pod creation: %v", err)
+	}
+
+	return pod, "", nil
+}
+
+// create returns a pod built from an index image or returns if an existing pod with the same name is found
+func (rp *RegistryPod) create(ctx context.Context) (*corev1.Pod, error) {
+	// call podForBundleRegistry() to make the pod definition
+	pod, err := rp.podForBundleRegistry()
+	if err != nil {
+		return nil, fmt.Errorf("error in building registry pod: %v", err)
+	}
+
+	// Check if Pod already exists
+	tmpPod, err := rp.Kubeclient.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			// Create pod in kubernetes cluster using the above pod definition and clientset
+			pod, err = rp.Kubeclient.CoreV1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{})
+			if err != nil {
+				return nil, fmt.Errorf("error creating registry pod: %v", err)
+			}
+		} else {
+			return nil, fmt.Errorf("error getting existing registry pod: %v", err)
+		}
+	} else {
+		return tmpPod, nil
+	}
+
+	return pod, nil
+}
+
+func (rp *RegistryPod) verifyPodRunning(ctx context.Context, pod *corev1.Pod) error {
+	// Upon creation of new pod, poll and verify that pod status is running
+	podCheck := wait.ConditionFunc(func() (done bool, err error) {
+		p, err := rp.Kubeclient.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error getting pod %s: %w", pod.Name, err)
+		}
+		return p.Status.Phase == corev1.PodRunning, nil
+	})
+
+	err := wait.PollImmediateUntil(time.Duration(200*time.Millisecond), podCheck, ctx.Done())
+	if err != nil {
+		return fmt.Errorf("error waiting for registry pod %s to run: %v", pod.Name, err)
+	}
+
+	return err
+}
+
+func (rp *RegistryPod) validate() error {
+	if len(rp.BundleImage) == 0 {
+		return errors.New("bundle image cannot be empty")
+	}
+	if len(rp.DBPath) == 0 {
+		return errors.New("registry database path cannot be empty")
+	}
+
+	if len(rp.Namespace) == 0 {
+		return errors.New("pod namespace cannot be empty")
+	}
+
+	if len(rp.BundleAddMode) == 0 {
+		panic("bundle add mode cannot be empty")
```

Comment:
There should be no reachable panics in user-facing code:
```suggestion
		return errors.New("bundle add mode cannot be empty")
```

Reply:
@joelanford  and I had a discussion about this. And the reasoning why he mentioned it should panic is that, if we error'ed out, that would be back up to the user who would expect to be able to change something to resolve the situation. But in this case, we always expect it to be set and if it is not, then there's a logic bug in the code, which has nothing to do with user input for this field.

I'm ok to change this back error out instead of panic, unless there are any strong objections.

Reply:
Shouldn't this field being defaulted to `replaces` remove the need for a panic?

Reply:
I agree with @estroz suggestion

Reply:
good point @estroz, reverting to error out instead of panic.